packages/components/src/internal/components/labels/PrintLabelsModal.tsx

```diff
@@ -1,28 +1,28 @@
 import React, { PureComponent } from 'react';
 import { Button, Modal } from 'react-bootstrap';
+import { List } from 'immutable';
 import { ComponentsAPIWrapper, getDefaultAPIWrapper } from '../../APIWrapper';
 
 import { HelpLink } from '../../util/helpLinks';
 import { QuerySelect } from '../forms/QuerySelect';
-import { SchemaQuery } from '../../../public/SchemaQuery';
 import { LabelHelpTip } from '../base/LabelHelpTip';
 import { Alert } from '../base/Alert';
 import { LoadingSpinner } from '../base/LoadingSpinner';
 
 import { InjectedQueryModels, withQueryModels } from '../../../public/QueryModel/withQueryModels';
+import { QueryModel } from '../../../public/QueryModel/QueryModel';
 
 import { BarTenderResponse } from './models';
 import { BAR_TENDER_TOPIC, LABEL_NOT_FOUND_ERROR } from './constants';
 
-interface Props {
+export interface PrintModalProps {
     afterPrint?: (numSamples: number, numLabels: number) => void;
     api?: ComponentsAPIWrapper;
     labelTemplate: string;
     onCancel?: (any) => void;
+    model?: QueryModel; // must provide either a model or schemaName/queryName
```

Comment:
nit: This comment doesn't make sense. A `schemaName/queryName` cannot be provided any longer. I would just remove this comment all together.

Second, is `model` really optional? It looks like all the usages in this PR (which is possibly all that exist in the world) are passing in a model. If so, I would mark this as required and then remove any defensive checks within the `PrintModal` implementation.

Lastly, please lint before merging.